src/projectscene/qml/Audacity/ProjectScene/trackspanel/TracksTitleBar.qml

```diff
@@ -22,11 +27,63 @@ KDDW.TitleBarBase {
     implicitHeight: gripButton.implicitHeight
     heightWhenVisible: expectedHeight
 
+    Component.onCompleted: {
+        if (effectsPanelWidth == 0) {
+            console.warn("effectsPanelWidth is not set ; doing some guesswork")
+            effectsPanelWidth = 240
```

Comment:
Can you please explain what's going on here?

Reply:
Yes:
`ProjectPage` constructs a `TracksTitleBar` (**A**) and a `TracksPanel` (**B**). Before this change, these two only had to fill the width of their parent.
Now, optionally, **A'** and **B'** will appear, meaning `TracksTitleBar` and `TracksPanel` need to use the same width so the end result looks like a regular grid.
![Unbenannt](https://github.com/user-attachments/assets/cacfd62f-df58-4e15-820f-46088a3c60a2)

In C++, I'd probably just add a non-optional ctor argument to these classes, and this would be passed by `ProjectPage`.
But I don't know of non-optional ctor args in QML. These lines you've commented were a feeble attempt from me to solve this: `ProjectPage`, when instantiating these two guys, would have to specify `TracksTitleBar { effectsPanelWidth: 240 }` and likewise for `TracksPanel`, and if it doesn't, I issue a warning.
I think the proper way would actually be to have a global constant for these two guys to reuse, but I thought you and Greg might teach me some best practice during the review?

Reply:
The title bar width is calculated based on the parent width (`anchors.fill: parent` for root), so it looks like you don't need to calculate anything extra here. 
try without these manipulations

Reply:
Yes, the title bar width is calculated based on the parent width, but then internally it has to know the relative width of its two sections, so I'm afraid there's no way around these manipulations.